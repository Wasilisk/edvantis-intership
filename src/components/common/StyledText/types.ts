import {StyledTextType} from "../../../models/common";

export type StyledTextProps = {
    content: StyledTextType
}