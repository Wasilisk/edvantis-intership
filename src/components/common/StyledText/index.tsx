import React from 'react';
import {StyledTextProps} from "./types";

const StyledText = ({content}: StyledTextProps) => {
    return (
        <>
            {content.before_highlight && <>{content.before_highlight}&nbsp;</>}
            {
                content.highlight &&
                <span className="highlight">{content.highlight}&nbsp;</span>
            }
            {content.after_highlight && content.after_highlight}
        </>
    );
};

export default StyledText;