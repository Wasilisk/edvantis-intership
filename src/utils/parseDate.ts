const month = ["January","February","March","April","May","June","July","August","September","October","November","December"];

export const parseDate = (timestamp: number) => {
    const date = new Date(timestamp);

    return `${month[date.getMonth()]} ${('0' + date.getDate()).slice(-2)}, ${date.getFullYear()}`;
}