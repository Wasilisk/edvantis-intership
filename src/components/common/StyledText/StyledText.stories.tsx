import {Story} from "@storybook/react";
import React from "react";
import {StyledTextType} from "../../../models/common";
import StyledText from "./index";

export default {
    title: 'Application/StyledText',
    component: StyledText,
    decorators: [
        (Story: Story) => (
            <h2>
                <Story/>
            </h2>
        )
    ]
};

const Template: Story<StyledTextType> = (props) => <StyledText {...props}/>;

export const SimpleText = Template.bind({});
SimpleText.args = {
    before_highlight: "Simple Text"
};

export const WithHighlight = Template.bind({});
WithHighlight.args = {
    before_highlight: "Text with",
    highlight: "Highlight"
};