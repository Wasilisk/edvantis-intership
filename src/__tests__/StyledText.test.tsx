import {render} from "./utils/store-wrapper";
import StyledText from "../components/common/StyledText";
import {mockStyledText} from "./__mock__/mock-styled-text";

describe('StyledText component', () => {
    it('should render component', () => {
        const {container, asFragment} = render(<p><StyledText {...mockStyledText}/></p>);

        expect(container.firstChild).toHaveTextContent(mockStyledText.before_highlight!);
        expect(container.firstChild).toHaveTextContent(mockStyledText.highlight!);
        expect(container.firstChild).toHaveTextContent(mockStyledText.after_highlight!);

        expect(asFragment()).toMatchSnapshot();
    })
})