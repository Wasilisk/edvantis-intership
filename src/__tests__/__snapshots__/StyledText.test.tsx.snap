// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText component should render component 1`] = `
<DocumentFragment>
  <p>
    Text before 
    <span
      class="highlight"
    >
      Highlight text 
    </span>
    Text after
  </p>
</DocumentFragment>
`;
